src/_guides/language/effective-dart/design.md

```diff
@@ -581,13 +582,41 @@ doesn't do that, it's best to assume you should *not* extend the class.
 Otherwise, later changes to it may break your code.
 
 
-### DO document whether your class supports being extended.
+### DO document if your class supports being extended.
 
 This is the corollary to the above rule. If you want to allow subclasses of your
 class, state that. Suffix the class name with `Base`, or mention it in the
 class's doc comment.
 
 
+### AVOID implementing a class that isn't intended to be an interface.
+
+Implicit interfaces are a powerful tool in Dart to avoid having to repeat the
+contract of a class when it can be trivially inferred from the signatures of an
+implementation of that contract.
+
+But implementing a class's interface is a very tight coupling to that class. It
+means virtually *any* change to the class whose interface you are implementing
+will break your implementation. For example, adding a new member to a class is
+usually a safe, non-breaking change. But if you are implementing that class's
+interface, now your class has a static error because it lacks an implementation
+of that new method.
+
+Package maintainers need the ability to evolve existing classes without breaking
+users. Treating *every* class as exposing an interface you are free to implement
+makes that very difficult for them. That difficulty in turn means the packages
+you rely on are slower to grow and adapt to new needs.
+
+To give the authors of the classes you use more leeway, avoid implementing their
```

Comment:
Unclear what "their" refers to. How about:

To give the authors of the classes you use more leeway, avoid implementing implicit interfaces except for classes that are clearly intended to be implemented.

Reply:
Done.